kernel/kernel-api/src/main/java/io/delta/kernel/data/VariantValue.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.data;
+
+/**
+ * Abstraction to represent a single Variant value in a {@link ColumnVector}.
+ */
+public interface VariantValue {
+    byte[] getValue();
```

Comment:
This may need a design decision. Is there a better interface we could provide that hides the complexity of the knowing `metadata` format? 

(... rough idea ... - haven't thought about all details)
- API to expose the paths (and potentially type of each path as well)
- API to get value. If the path is an int `VariantValue.getInt(String path) -> return integer`.